Assets/Scripts/Layers/LayerTypes/HierarchicalObjectLayerGameObject.cs

```diff
@@ -68,62 +75,72 @@ protected void OnDestroy()
 
         private void UpdatePosition(Coordinate newPosition)
         {
-            if (newPosition.ToUnity() != transform.position)
-                transform.position = newPosition.ToUnity();
+            if (newPosition.ToUnity() == worldTransform.Coordinate.ToUnity()) return;
+
+            worldTransform.Coordinate = newPosition;
+            transform.position = worldTransform.Coordinate.ToUnity();
         }
 
         private void UpdateRotation(Vector3 newAngles)
         {
-            if (newAngles != transform.eulerAngles)
-                transform.eulerAngles = newAngles;
+            if (newAngles == transform.eulerAngles) return;
+
+            worldTransform.Rotation = Quaternion.Euler(newAngles);
+            transform.rotation = worldTransform.Rotation;
         }
 
         private void UpdateScale(Vector3 newScale)
         {
-            if (newScale != transform.localScale)
-                transform.localScale = newScale;
+            if (newScale == transform.localScale) return;
+
+            transform.localScale = newScale;
         }
 
         public virtual void LoadProperties(List<LayerPropertyData> properties)
         {
-            var transformProperty = (TransformLayerPropertyData)properties.FirstOrDefault(p => p is TransformLayerPropertyData);
-            if (transformProperty != null)
+            var transformProperty = properties.OfType<TransformLayerPropertyData>().FirstOrDefault();
+            if (transformProperty == default(TransformLayerPropertyData)) return;
```

Comment:
waarom niet expliciet `transformProperty == null` hier?

Reply:
Als je in `properties.OfType<TransformLayerPropertyData>().FirstOrDefault()` 'ofType' gebruikt dan heb je het voordeel dat de conversie naar TransformLayerPropertyData automatisch goed gaat, dus heb je geen casting meer nodig, maar het betekent wel dat FirstOrDefault niet perse een null value teruggeeft, maar de 'default' van TransformLayerPropertyData. Dus als je FirstOrDefault gebruikt, dan is het veiliger om `default(...)` te gebruiken zodat je zeker weet dat de check klopt (en default is regelmatig null, maar niet altijd; zoals bij structs IIRC)

Reply:
ja, default is alleen null als de Type nullable is.

maar voor mij geeft een `== null` aan dat er niks is gevonden, en hoe ik dit lees is dat er een default waarde is gevonden, wat niet per se null hoeft te zijn.

verder is dit geen blokkerende issue, maar ik vroeg me gewoon af wat het voordeel hiervan was, al begrijp ik jouw uitleg nog niet helemaal

Reply:
FirstOrDefault geeft geen null terug als die niks vind, die geeft -zoals de methode naam aangeeft- de default terug als die niks vind. Toevallig is dat voor nullable type 'null'; maar het is met FirstOrDefault vaak beter -zeker in combinatie met ofType omdat dan default daadwerkelijk over die type gaat- om te checken op `default(type)`.

Het verschil zit hem in dit `ofType` als je die niet gebruikt, dan weet Linq niet welk type resultaat eruit gaat komen, dus dan is default 'null'. Maar als je ofType gebruikt, dan krijg je automatische typecasting, maar dus ook van je default; en dus moet je checken op default(type) en niet op null